pkg/resource/azurepathfixjob.go

```diff
@@ -269,11 +271,39 @@ func (gapfj *generatorAzurePathFixJob) Create() (runtime.Object, error) {
 }
 
 func (gapfj *generatorAzurePathFixJob) Update(o runtime.Object) (runtime.Object, bool, error) {
-	return commonUpdate(gapfj, o, func(obj runtime.Object) (runtime.Object, error) {
-		return gapfj.client.Jobs(gapfj.GetNamespace()).Update(
-			context.TODO(), obj.(*batchv1.Job), metav1.UpdateOptions{},
-		)
-	})
+	// updating jobs doesn't work like other objects - we get validation errors
+	// if we try. in our case, we only care about the job container's env vars,
+	// so we check if the existing job's container env vars match the expected,
+	// and if they don't we recreate the job.
+	exp, err := gapfj.expected()
+	if err != nil {
+		return nil, false, err
+	}
+	expectedJob := exp.(*batchv1.Job)
```

Comment:
nit: is it better to check the conversion with `expectedJob, ok` to ignore the ones that is `!ok`